### PR TITLE
feat: toggle settlement health event upon settlement batch submission status

### PR DESCRIPTION
**Pull request by omritoptix**
## Overview
This pr moves the batch submission to settlement layer logic from the manager to the settlement.
By that it allows us to encapsulate the logic and handle any possible failures.
The manager will only be notified upon submission success or health status changes. 
 
# PR Standards

## Opening a pull request should be able to meet the following requirements

---

For Author:

- [x]  Targeted PR against correct branch
- [x]  included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [x]  Linked to Github issue with discussion and accepted design
- [x]  Targets only one github issue
- [x]  Wrote unit and integration tests
- [ ]  All CI checks have passed
- [ ]  Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code)

---

For Reviewer:

- [ ]  confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Reviewers assigned
- [ ]  confirmed all author checklist items have been addressed

---

After reviewer approval:

- [ ]  In case targets main branch, PR should be squashed and merged.
- [ ]  In case PR targets a release branch, PR should be rebased.

### CODE REVIEW

File: block/manager.go [hunk 1]
```diff
@@ -720,7 +708,7 @@ func (m *Manager) submitNextBatch(ctx context.Context) {
 	// Submit batch to SL
 	// TODO(omritoptix): Handle a case where the SL submission fails due to syncTarget out of sync with the latestHeight in the SL.
 	// In that case we'll want to update the syncTarget before returning.
-	m.submitBatchToSL(nextBatch, resultSubmitToDA)
+	go m.settlementClient.SubmitBatch(nextBatch, m.dalc.GetClientType(), resultSubmitToDA)
```
Comment (mtsitrin) [inline block/manager.go:711]: U have a different style with the DA layer and the SL layer
The DA layer has `m.submitBatchToDA` which manages the retries and the result
While now the SL layer is managed entirely from `m.settlementClient`

File: settlement/base.go [hunk 2]
```diff
@@ -93,27 +93,11 @@ func (b *BaseLayerClient) Stop() error {
 	return nil
 }
 
-// SubmitBatch submits the batch to the settlement layer. This should create a transaction which (potentially)
-// triggers a state transition in the settlement layer.
-func (b *BaseLayerClient) SubmitBatch(batch *types.Batch, daClient da.Client, daResult *da.ResultSubmitBatch) *ResultSubmitBatch {
+// SubmitBatch tries submitting the batch in an async way to the settlement layer. Events are emitted on success or failure.
```
Comment (mtsitrin) [inline settlement/base.go:96]: How is that `async`?
You still call `d.client.BroadcastTx` and waits to see the tx result

File: settlement/base.go [hunk 3]
```diff
@@ -159,14 +143,13 @@ func (b *BaseLayerClient) fetchSequencersList() ([]*types.Sequencer, error) {
 	return sequencers, nil
 }
 
-func (b *BaseLayerClient) validateBatch(batch *types.Batch) error {
+func (b *BaseLayerClient) validateBatch(batch *types.Batch) {
```
Comment (mtsitrin) [inline settlement/base.go:146]: Not sure why to remove the `error`
I don't think it's good idea to have panic scattered around

The internal functions should return errors, to be aggregated and panic upon if needed

File: settlement/dymension/dymension.go [hunk 4]
```diff
@@ -262,26 +345,28 @@ func (d *HubClient) eventHandler() {
...
-		return nil, err
+		d.logger.Error("Error getting account for building transaction", "account", d.config.DymAccountName)
+		panic(err)
 	}
 
 	addr, err := account.Address(addressPrefix)
 	if err != nil {
-		return nil, err
+		d.logger.Error("Error getting address prefix for building transaction", "account", addressPrefix)
+		panic(err)
```
Comment (mtsitrin) [inline settlement/dymension/dymension.go:369]: same here
I guess this error could be checked by the caller and panic

**Review by mtsitrin — APPROVED** [review]
I don't like the excessive usage of `panic` instead of returning `errors`,
IMO it's bad practice

File: block/manager.go (hunk 1 above)
Comment (omritoptix) [reply block/manager.go:711]: true. that's the [next task ](https://github.com/dymensionxyz/dymint/issues/314) to move the DA submission logic there.

**Comment by omritoptix** [comment]
> I don't like the excessive usage of `panic` instead of returning `errors`, IMO it's bad practice

I see. don't think it's about liking or not nor that it's a bad practice. I it's about the use case for panic. 
For me panicing is in a case where condition is met that cannot be or should not be handled, such as a bug in the code which the process can't recover from.

But let me know which places specifically you think I should have used errors instead of panic so can I check.

File: settlement/base.go (hunk 2 above)
Comment (omritoptix) [reply settlement/base.go:96]: it's async by the [cosmos terms](https://tutorials.cosmos.network/academy/2-cosmos-concepts/3-transactions.html). submitting a tx in a broadcast mode async in cosmos means submitting the tx and not waiting to see if it was included in a block (but do get a response back). I can make it more clear that it's broadcast mode = async if you think it makes more sense.

File: settlement/base.go (hunk 3 above)
Comment (omritoptix) [reply settlement/base.go:146]: makes sense

**Review by ItayLevyOfficial — APPROVED** [review]

**Pull request merged**